reflections.go

```diff
@@ -91,6 +91,26 @@ func GetFieldTag(obj interface{}, fieldName, tagKey string) (string, error) {
 	return field.Tag.Get(tagKey), nil
 }
 
+// Function to get the the Field from it's json tag
+func GetNameFieldByTag(obj interface{}, tag string, key string) (string, error) {
```

Comment:
I would prefer this function to be called `GetFieldNameByTagValue` instead. Moreover, I would find it more intuitive if the two last arguments were inverted and renamed for clarity:

```
func GetFieldNameByTagValue(obj interface{}, tagKey, tagValue string) (string, error)
```